src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToForEachProvider.cs

```diff
@@ -0,0 +1,769 @@
+
+                var returnType = methodSymbol.ReturnType;
+                if (returnType.Name != "IEnumerable") // TODO better that name check?
+                {
+                    return false;
+                }
+
+                var typeArguments = returnType.GetTypeArguments();
+                if (typeArguments.Length != 1)
+                {
+                    return false;
+                }
+
+                var typeArgument = typeArguments.Single();
+                switch (methodSymbol.Parameters.Length)
+                {
+                    // IEnumerable<T> Where(Func<T, bool> predicate) 
+                    case 1:
+                        return IsPredicate(methodSymbol.Parameters.First().Type, typeArgument);
+
+                    // IEnumerable<T> Where(this IEnumerable<T>, Func<T, bool> predicate) 
+                    case 2:
+                        // TODO add checks for this.
+                        return IsPredicate(methodSymbol.Parameters.First().Type, typeArgument);
+
+                    default: return false;
+                }
+            }
+
+            private bool TryGetMethodSymbol(QueryClauseSyntax queryClause, out IMethodSymbol methodSymbol)
+            {
+                QueryClauseInfo queryClauseInfo = _semanticModel.GetQueryClauseInfo(queryClause, _cancellationToken);
+                methodSymbol = queryClauseInfo.OperationInfo.Symbol as IMethodSymbol;
+                return methodSymbol != null;
+            }
+
+            private static bool IsPredicate(ITypeSymbol typeSymbol, ITypeSymbol typeArgument)
+            {
+                if (typeSymbol.Name != "Func")  // TODO better that name check?
+                {
+                    return false;
+                }
+
+                var typeArguments = typeSymbol.GetTypeArguments();
+                if (typeArguments.Length != 2)
+                {
+                    return false;
+                }
+
+                return typeArguments.Last()?.SpecialType == SpecialType.System_Boolean &&
+                    SignatureComparer.Instance.HaveSameSignature(typeArguments.First(), typeArgument, caseSensitive: true);
+            }
+
+            private bool TryProcessFromClause(FromClauseSyntax fromClause, ProcessingState processingState)
+            {
+                QueryClauseInfo queryClauseInfo = _semanticModel.GetQueryClauseInfo(fromClause, _cancellationToken);
+                if (queryClauseInfo.OperationInfo.Symbol == null || // First FromClause may have no OperationInfo symbols.
+                    (queryClauseInfo.OperationInfo.Symbol is IMethodSymbol methodSymbol && methodSymbol.Parameters.Length == 2))
+                {
+                    processingState.AddVariableName(fromClause.Identifier.ValueText);
+                    processingState.AddStatementProcessingMethod(
+                        s => SyntaxFactory.ForEachStatement(
+                                    fromClause.Type ?? VarNameIdentifier,
```

Comment:
it concerns me we will use 'var' when some people do not want that, and have set their code style options to never prefer 'var'.

Reply:
I will definitely fix this. :-) [https://github.com/dotnet/roslyn/issues/25639](https://github.com/dotnet/roslyn/issues/25639)

---
In reply to: [176869164](https://github.com/dotnet/roslyn/pull/25362#discussion_r176869164) [](ancestors = 176869164)